managed/CounterStrikeSharp.API/Modules/Commands/Targeting/Target.cs

```diff
@@ -80,48 +83,36 @@ public Target(string target)
-            case TargetType.TeamT:
-                return player.TeamNum == (byte)CsTeam.Terrorist;
-            case TargetType.TeamSpec:
-                return player.TeamNum == (byte)CsTeam.Spectator;
-            case TargetType.GroupAll:
-                return !player.IsHLTV;
-            case TargetType.GroupBots:
-                return player.IsBot;
-            case TargetType.GroupHumans:
-                return !player.IsBot && !player.IsHLTV;
-            case TargetType.GroupAlive:
-                return player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_ALIVE };
-            case TargetType.GroupDead:
-                return player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_DEAD or (byte)LifeState_t.LIFE_DYING };
-            case TargetType.GroupNotMe:
-                return player.SteamID != caller?.SteamID;
-            case TargetType.PlayerMe:
-                return player.SteamID == caller?.SteamID;
-            case TargetType.IdUserid:
-                return player.UserId.ToString() == Slug;
-            case TargetType.IdSteamEscaped:
-                return ((SteamID)player.SteamID).SteamId2 == Slug;
-            case TargetType.IdSteam64:
-                return ((SteamID)player.SteamID).SteamId64.ToString() == Slug;
-            case TargetType.ExplicitName:
-            case TargetType.ImplicitName:
-                return player.PlayerName.Contains(Slug, StringComparison.OrdinalIgnoreCase);
-            default:
-                return false;
-        }
+            TargetType.PlayerAim => caller != null && player == gameRules.GetClientAimTarget(caller),
+            TargetType.TeamCt => player.Team == CsTeam.CounterTerrorist,
+            TargetType.TeamT => player.Team == CsTeam.Terrorist,
+            TargetType.TeamSpec => player.Team == CsTeam.Spectator,
+            TargetType.GroupAll => !player.IsHLTV,
+            TargetType.GroupBots => player.IsBot,
+            TargetType.GroupHumans => !player.IsBot && !player.IsHLTV,
+            TargetType.GroupAlive => player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_ALIVE },
+            TargetType.GroupDead => player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_DEAD or (byte)LifeState_t.LIFE_DYING },
+            TargetType.GroupNotMe => player.SteamID != caller?.SteamID,
+            TargetType.PlayerMe => player.SteamID == caller?.SteamID,
+            TargetType.IdUserid => player.UserId.ToString() == Slug,
+            TargetType.IdSteamEscaped => ((SteamID)player.SteamID).SteamId2 == Slug,
+            TargetType.IdSteam64 => ((SteamID)player.SteamID).SteamId64.ToString() == Slug,
+            TargetType.ExplicitName => player.PlayerName.Contains(Slug, StringComparison.OrdinalIgnoreCase),
+            TargetType.ImplicitName => player.PlayerName.Contains(Slug, StringComparison.OrdinalIgnoreCase),
+            _ => false
+        };
```

Comment:
are we sure we want to switch this whole part to this syntax?

Reply:
It looks better imo

Reply:
> It looks better imo

well, thats preference

I wonder though, can we set the gamerules nullable and have `null` as default value? I know thats its a private method but maybe only get and pass gamerules if the target type requires it